gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+        mock_load_yaml.side_effect = yaml_side_effect2
+        self.assertEqual(configuration_tests.check_stata(), None)
+
+        # Failures
+
+        # No yaml value and no default value in path
+        mock_stata_exec.side_effect = lambda x: None
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_stata()
+
+        # Bad package
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_stata(packages = ['bad_package'])
+
+
+    @mock.patch('gslab_scons.configuration_tests.raw_input')
+    def test_load_yaml_value(self, mock_raw_input):
+        # Setup
+        def raw_input_side_effect(*args, **kwargs):
+            message = args[0]
+            if re.search("corrupted", message):
+                return 'y'
+            if re.search("Enter", message):
+                return 'value'
+
+        def raw_input_side_effect2(*args, **kwargs):
+            message = args[0]
+            if re.search("corrupted", message):
+                return 'n'
+            if re.search("Enter", message):
+                return 'none'
+
+        mock_raw_input.side_effect = raw_input_side_effect
+
+        def make_yaml(string):
+            if os.path.isfile('yaml.yaml'):
+                os.remove('yaml.yaml')  
+            with open('yaml.yaml', 'wb') as f:
+                f.write('%s\n' % string)
+
+        # Test Good
+
+        # Yaml file exists, is not corrupt, and has key.
+        make_yaml('key: value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file exists, is not corrupt, and doesn't have key. User enters value for key.
+        make_yaml('bad_key: bad_value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+        
+        # But now yaml has a correct key and doesn't require user input.
+        mock_raw_input.side_effect = lambda x: "bad_value"
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+        mock_raw_input.side_effect = raw_input_side_effect
+
+        # Yaml file exists and is corrupt. User deletes and re-creates. 
+        make_yaml('key value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters value to create. 
+        os.remove('yaml.yaml')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters none value to create. 
+        os.remove('yaml.yaml')
+        mock_raw_input.side_effect = raw_input_side_effect2
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Yaml file now exists with none value.
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
```

Comment:
This would be true even if the file didn't exist (i.e. this line is the same as the previous assert statement). Maybe we could check if the yaml file exists directly?